### ISSUE TITLE
`setup.py install` does not install headers

### ISSUE THREAD

**Issue by giacomov** [healpy/healpy#404]
If I do `python setup.py install ` the headers of the CXX version of healpix are not installed, even though the library is compiled during the setup.

If I do `python setup.py build` and then check the `build` directory, I see that the headers are in the `temp*` folder:

```
(work_env) ~/.../healpy/healpy-1.10.3/build > cd temp.linux-x86_64-2.7
(work_env) ~/.../healpy-1.10.3/build/temp.linux-x86_64-2.7 > find . -name "healpix_base.h"
./include/healpix_cxx/healpix_base.h
```
but not in the `lib.*` folder:

```
(work_env) ~/.../healpy/healpy-1.10.3/build > cd lib.linux-x86_64-2.7/
(work_env) ~/.../healpy-1.10.3/build/lib.linux-x86_64-2.7 > find . -name "healpix_base.h"
```
The headers are necessary to build libraries against healpix. 

This issue propagate to the `conda` installation of healpy from conda-forge as well, since the conda recipe runs the setup.py.

**Comment by zonca** [comment]
we generally assume that the HEALPix version is only used internally by `healpy`.

@mreineck do you see any issue if we install the HEALPix C++ headers?

**Comment by lpsinger** [comment]
> @mreineck do you see any issue if we install the HEALPix C++ headers?

There is no point in installing the headers because we statically link the HEALPix C++ library if we are building from the bundled sources.

**Comment by zonca** [comment]
so we're not creating dynamic libraries?

**Comment by lpsinger** [comment]
Nope. Why would we?

**Comment by zonca** [comment]
I see, we create dynamic libraries of the healpy extensions that are loaded
by Python (all the `*.so` inside the healpy installed folder) and those
embed statically the HEALPix libraries. So no standalone HEALPix libraries
are made available.
To have those you need to install HEALPix separately.

**Comment by lpsinger** [comment]
> If not, we'll just go ahead and create a conda package for the C++ part of healpix on its own.

That would be the better solution. It would be awesome if you could update the healpy conda package to use said shared library conda package. Please use the tarball `healpix_cxx-3.30.0.tar.gz` from https://sourceforge.net/projects/healpix/files/Healpix_3.30/.

**Comment by giacomov** [comment]
ok, I'll let you know when we manage to do it.

> Please use the tarball healpix_cxx-3.30.0.tar.gz

Is there a reason not to use 3.31?

**Comment by lpsinger** [comment]
> ok, I'll let you know when we manage to do it.
> 
> Please use the tarball healpix_cxx-3.30.0.tar.gz
> Is there a reason not to use 3.31?

Yeah, because the autotools tarball has not been generated for it yet.

**Comment by mreineck** [comment]
> Yeah, because the autotools tarball has not been generated for it yet.

There is no need for a new tarball ... the C++ part did not change noticeably between 3.30 and 3.31.

**Comment by giacomov** [comment]
I generated the conda package for the C++ library. I ended up using v3.31 because it corrects the bug in query_disc which might be important for our applications.

The conda package is here:

https://anaconda.org/giacomov/healpix_cxx

The recipe and scripts to build the package are here:

https://github.com/giacomov/healpix-cxx-conda

Comments are welcome. At some point it would be nice to have that into conda-forge, but I'll wait input from the authors first.

**Comment by giacomov** [comment]
I just created the recipe on conda-forge:


It would probably be best if I could add some of the authors to the conda-forge recipe, once it is merged, so you have the authority to make changes as well.

**Comment by zonca** [comment]
we had it already: https://github.com/conda-forge/healpy-feedstock

**Comment by giacomov** [comment]
@zonca sorry, this conversation is about the C++ part of Healpix, not healpy. The new recipe contains the C++ libraries.

**Comment by zonca** [comment]
@giacomov you first need approval from the Healpix team, please contact them on the mailing lists on their website: https://sourceforge.net/p/healpix/mailman/

**Comment by giacomov** [comment]
@zonca thanks, I sent them a private message a few hours ago. Let's see what they answer.

**Comment by zonca** [comment]
if you get a positive answer please post a link to the feedstock on this
thread

**Comment by mreineck** [comment]
Sorry for getting into this discussion late...

For what exactly do you need Healpix maintainer approval?

**Comment by giacomov** [comment]
Even though I don't think I need approval  - the software is released under GPL, so I can redistribute it under the same license - it would be best if they are onboard, and it is for sure a courtesy to ask.

**Comment by mreineck** [comment]
I'm personally fine with it; thanks for asking!
